builder/dockerfile/internals.go

```diff
@@ -383,11 +383,9 @@ func (b *Builder) calcCopyInfo(cmdName, origPath string, allowLocalDecompression
 		hfi.SetHash("file:" + hfi.Hash())
 		return copyInfos, nil
 	}
-
 	// Must be a dir
-
 	var subfiles []string
-	b.context.Walk(origPath, func(path string, info builder.FileInfo, err error) error {
+	b.context.Walk(statPath, func(path string, info builder.FileInfo, err error) error {
```

Comment:
Error handling has been added. I had to switch to a filesystem backed walker because otherwise there could be errors on walking the context if files had been removed with `context.Remove()`.